### PR TITLE
Refresh REST API

### PR DESCRIPTION
**Pull request by simondeziel**
When reviewing the weekly news, I noticed the `create_at` field for `storage volume` and `storage volume snapshot` was missing from #11034. I then went on manually adding them to `doc/rest-api.yaml` only to notice it was auto-generated (good!) and was missing some other bits. To make it extra clear (for the future me), I updated the doc while in there.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -107,9 +107,7 @@ update-schema:
 
 .PHONY: update-api
 update-api:
-ifeq "$(LXD_OFFLINE)" ""
-	(cd / ; go install -v -x github.com/go-swagger/go-swagger/cmd/swagger@latest)
-endif
+	command -v swagger >/dev/null || (cd / ; go install -v -x github.com/go-swagger/go-swagger/cmd/swagger@latest)
```
Comment (tomponline) [inline Makefile:110]: Why did you remove the offline check?

**Comment by tomponline** [comment]
Thanks lgtm, just need to understand the change to swagger command logic.

File: Makefile (hunk 1 above)
Comment (tomponline) [reply Makefile:110]: if its missing and in offline mode wont this now fail?

Comment (simondeziel) [reply Makefile:110]: Oh that's right, I didn't properly understood the `ifeq` but in that case, if it's missing and in offline mode, wouldn't the next command fail anyway?

Comment (monstermunchkin) [reply Makefile:110]: The way I understand it, is that if `LXD_OFFLINE` is set, we should only do offline stuff. That includes _not_ running `go install`.

This commit ignores the offline mode as it would pull swagger if it's not available. I don't believe we should do this. Another issue is that it would never update swagger if it's already installed.

**Review by monstermunchkin — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (tomponline) [reply Makefile:110]: also if swagger is installed but is out of date it would get updated but with this change it would not. i think we should drop this commit.

**Review by monstermunchkin — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (simondeziel) [reply Makefile]: > The way I understand it, is that if `LXD_OFFLINE` is set, we should only do offline stuff. That includes _not_ running `go install`.
> 
> This commit ignores the offline mode as it would pull swagger if it's not available. I don't believe we should do this. 

Irrespective of the `LXD_OFFLINE` mode, if swagger is not available, the `swagger generate` would have failed, no?

> Another issue is that it would never update swagger if it's already installed.

I wasn't aware that `go install` would update it, thanks. I've dropped the commit.

Comment (tomponline) [reply Makefile]: Yes it would have failed

**Review by tomponline — APPROVED** [review]

File: test/lint/rest-api-up-to-date.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+f="doc/rest-api.yaml"
+
+rest_api_hash() {
+  md5sum "${f}" | cut -f1 -d" "
+}
+
+echo "Checking that ${f} is up to date..."
+
+# make sure the .pot is updated
```
Comment (tomponline) [inline test/lint/rest-api-up-to-date.sh:11]: I think this is a copy paste error?
